src/renderer/lib/state.js

```diff
@@ -133,26 +132,25 @@ function setupStateSaved (cb) {
 
   const tasks = []
 
-  config.DEFAULT_TORRENTS.map(function (t, i) {
+  config.DEFAULT_TORRENTS.forEach((t, i) => {
     const infoHash = saved.torrents[i].infoHash
-    tasks.push(function (cb) {
+    tasks.push(
       cpFile(
         path.join(config.STATIC_PATH, t.posterFileName),
         path.join(config.POSTER_PATH, infoHash + path.extname(t.posterFileName))
-      ).then(cb).catch(cb)
-    })
-    tasks.push(function (cb) {
+      )
+    )
+    tasks.push(
       cpFile(
         path.join(config.STATIC_PATH, t.torrentFileName),
         path.join(config.TORRENT_PATH, infoHash + '.torrent')
-      ).then(cb).catch(cb)
-    })
+      )
+    )
   })
 
-  parallel(tasks, function (err) {
-    if (err) return cb(err)
-    cb(null, saved)
-  })
+  Promise.all(tasks)
+    .then(() => cb(null, saved))
+    .catch(err => cb(err))
```

Comment:
This is not the correct way to call an error callback with a promise. It's an easy mistake to make.

The `.catch` will catch more than just the promise rejections from the `tasks` array. It will also catch any exceptions thrown from the handler in `.then()` which means that if the `cb` function throws an exception, then we'll catch it here.

Instead we should use `.then(success, fail)`. I'll send a PR.

Reply:
PR here: https://github.com/webtorrent/webtorrent-desktop/pull/1679